### ISSUE TITLE
Zipfile created but not found

### ISSUE THREAD

**Issue by ceesjandronkers** [google-github-actions/deploy-cloud-functions#258]
### TL;DR

My google cloud function deployment gives an error on finding the generated zip-file. In the error message, it refers to a path which is not matching the created zip file.

### Expected behavior

I would expect that the zipped file would have been found and sent to the google cloud functions api.

### Observed behavior

First, the creation of the zipfile is logged:

```
Created zip file from './' at '/tmp/cfsrc-fe01cc8fcadfea8e32b25ac5.zip'
```

After that, I get the following error message (note the difference in the paths):

```
Error: google-github-actions/deploy-cloud-functions failed with: failed to upload zip file: The file at /home/runner/work/Altilium.Core/Altilium.Core/4c3724d897b81ccdb8c0296e does not exist, or it is not a file. ENOENT: no such file or directory, lstat '/home/runner/work/Altilium.Core/Altilium.Core/4c3724d897b81ccdb8c0296e'
```

### Action YAML

```yaml
name: CloudDeploy

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
    deploy:
        name: Deploy to GCloud
        runs-on: ubuntu-latest
        permissions:
          contents: 'read'
          id-token: 'write'
          
        steps:
        - name: Authenticate
          uses: google-github-actions/auth@v0
          with:
            credentials_json: ${{ secrets.GCP_CREDENTIALS }}
            service_account: ${{ secrets.GCP_EMAIL }}

        - name: Checkout repository
          uses: actions/checkout@v2
        
        - name: Deploy
          uses: 'google-github-actions/deploy-cloud-functions@v0'
          continue-on-error: true
          with:
            name: 'altilium-api'
            runtime: 'python39'
            region: 'europe-west1'
            entry_point: 'run_api.py'
            
        - name: Debug output of zip path
          run: 'ls /tmp'
          
        - name: Debug output of missing path
          run: 'ls /home/runner/work/Altilium.Core/Altilium.Core'

        # Example of using the output
        - id: 'test'
          run: 'curl "${{ steps.deploy.outputs.url }}"'
```

### Additional information

I am quite new to all this, so if the bug is in my YAML, please say so.

I checked if the ZIP-file was created, and it was.

I tried to trace what's happening in the code. As far as I can see, the error can only originate from [here](https://github.com/google-github-actions/deploy-cloud-functions/blob/938f2f686126c0e4ad3601df45408041b4650c6e/src/client.ts#L520). But I don't understand why fs.createReadStream would stat a file in the current directory when given a zip file with an absolute path.

**Comment by sethvargo** [comment]
1. Make sure you put `checkout` before `auth`.

2. What are the contents of `Altilium.Core`? Are there symlinks by chance?

**Comment by ceesjandronkers** [comment]
Suggestion one solved the problem, although I do not understand why the order of `checkout `and `auth` would influence opening the zip file. 

Thank you!

**Comment by sethvargo** [comment]
I think the error might be misleading. If you don't initiate a checkout before auth, there's nowhere to store the exported credentials.

**Comment by ceesjandronkers** [comment]
That's good to know. Thanks for the help, again.
